docs: recommended values for extensions.json (#46)
open-vsx/publish-extensions: README.md

      // Unique Open VSX extension ID in the form "<namespace>.<name>"
      "id": "redhat.vscode-yaml",
      // Repository URL to clone and publish from

>       "repository": "https://github.com/redhat-developer/vscode-yaml",
>       // (RECOMMENDED) The version to publish to Open VSX (defaults to the package.json version)
>       "version": "0.27.0",
>       // (RECOMMENDED) The Git branch, tag, or commit to check out before publishing (defaults to the repository's default branch)
>       "checkout": "v0.27.0",

Reviewer: Since repositories don't always have dedicated "release" branches or tags, I prefer keeping `checkout` optional, and also keeping the first example minimal (i.e. only required parameters, since the optional ones are documented below).

```suggestion
      "repository": "https://github.com/redhat-developer/vscode-yaml"
```

    {
      // Unique Open VSX extension ID in the form "<namespace>.<name>"
      "id": "rebornix.ruby",
      // Repository URL to clone and publish from
      "repository": "https://github.com/rubyide/vscode-ruby",

>       // (RECOMMENDED) The version to publish to Open VSX (defaults to the package.json version)

Reviewer: Calling it `RECOMMENDED` is a great idea, thanks! 👍 I also like the proposed new options order.

However, my original description was slightly wrong, so I might as well drive-by fix it. 😇

Explanation: This value is only used in comparison with what OpenVSX already has, to know if publishing should be skipped or not (but if publishing happens, it's the actual version string in `package.json` that will be used).

```suggestion
      // (RECOMMENDED) The version that should be published on Open VSX (defaults to the package.json version)
```

      "version": "0.27.0",

>       // (RECOMMENDED) The Git branch, tag, or commit to check out before publishing (defaults to the repository's default branch), should match "version"

Reviewer: Due to the reason described above, it's ok to not match "version" (this could also be a "release" or "stale" branch, and you can even omit "version", which just serves to skip publishing when it's not necessary).

```suggestion
      // (RECOMMENDED) The Git branch, tag, or commit to check out before publishing (defaults to the repository's default branch)
```
